docs/examples/controlmodels/control_models_inference_with_linear_gaussian_control_model.py

```diff
@@ -0,0 +1,330 @@
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# To define the control model class, we will create a variant of :class:`~.LinearControlModel` and
+#  ovewrite the :attr:`~.LinearControlModel.matrix` attribute according to the above expression.
+
+
+from stonesoup.models.control.linear import LinearControlModel
+
+
+class ConstantAccelerationLinearControlModel(LinearControlModel):
+    """A model that applies a constant acceleration over a specified time period.
+    This is just a :class:`~.LinearControlModel` which accepts a time_interval input
+    to matrix to compute the control matrix.
+
+    A location and velocity state vector with location and velocity interleaved is assumed.
+    """
+
+    def matrix(self, time_interval, **kwargs) -> np.ndarray:
+        r"""
+
+        Parameters
+        ----------
+        time_interval : :class:`datetime.timedelta`
+            A time interval. Note the units used are :math:`s` so accelerations are implicitly
+            per second squared.
+
+        Returns
+        -------
+        : :class:`numpy.ndarray`
+            the control-input model matrix, :math:`B_k`
+        """
+
+        # Calculate time interval
+        deltat = time_interval.total_seconds()
+        # Define B for one dimension
+        onedm = np.array([[(deltat**2)/2.0], [deltat]])
+        # Construct control matrix for each dimension
+        control_matrix = self.control_matrix
+        for i in range(0, self.ndim):
+            control_matrix[2*i:2*i+2, i:i+1] = onedm
+
+        self.control_matrix = control_matrix
+
+        return self.control_matrix
+
+
+# %%
+# Generate Ground Truth
+# ^^^^^^^^^^^^^^^^^^^^^
+# Now that the control model is defined, it can now be used to create the ground truth
+# for the target. There are two main differences compared to generating ground truth
+# without a control model. The first is deciding on control input which is required and
+# should be stored for use later in the filtering process. The second is we have to apply
+# the transition model function and the control model function when progressing states.
+#
+# In this example, the target will undergo two manoeuvres. The first starting at 40 seconds
+# and lasting 50 seconds which will be a turn to the right and the second manoeuvre will be
+# at 120 seconds, again lasting 50 seconds, and will be a turn to the left. Nominally, the
+# target will traverse according to nearly constant velocity when not manoeuvering.
+#
+# First we define the models and populate the required parameters.
+
+
+# Transition model
+from stonesoup.models.transition.linear import CombinedLinearGaussianTransitionModel, \
+                                               ConstantVelocity
+transition_model = CombinedLinearGaussianTransitionModel([ConstantVelocity(0.0005),
+                                                          ConstantVelocity(0.0005)])
+
+# Control model
+control_model = ConstantAccelerationLinearControlModel(np.array([[1., 0],
```

Comment:
Do we want to mention that the input control matrix here is a dummy just used to force the correct dimension? The custom class above overwrites the initial values with the ones we want.